examples/api/test/material/theme/theme_extension.1_test.dart

```diff
@@ -0,0 +1,110 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_api_samples/material/theme/theme_extension.1.dart'
+    as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('ThemeExtension can be obtained', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      const example.ThemeExtensionExampleApp(),
+    );
+
+    final ThemeData theme = Theme.of(
+      tester.element(find.byType(example.Home)),
+    );
+    final example.MyColors colors = theme.extension<example.MyColors>()!;
+
+    expect(colors.brandColor, equals(const Color(0xFF1E88E5)));
+    expect(colors.danger, equals(const Color(0xFFE53935)));
+  });
+
+  testWidgets('ThemeExtension can be changed', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      const example.ThemeExtensionExampleApp(),
+    );
+
+    ThemeData theme = Theme.of(
+      tester.element(find.byType(example.Home)),
+    );
+    example.MyColors colors = theme.extension<example.MyColors>()!;
+
+    expect(colors.brandColor, equals(const Color(0xFF1E88E5)));
+    expect(colors.danger, equals(const Color(0xFFE53935)));
+
+    await tester.tap(find.byType(IconButton));
```

Comment:
nit: add a comment indicating what the tap is doing, in this case I think switching the theme to dark mode.